src/Service/Startup.cs

```diff
@@ -535,6 +556,49 @@ private void ConfigureAuthentication(IServiceCollection services, RuntimeConfigP
             }
         }
 
+        /// <summary>
+        /// Configure Application Insights Telemetry based on the loaded runtime configuration. If Application Insights
+        /// is enabled, we can track different events and metrics.
+        /// </summary>
+        /// <param name="runtimeConfigurationProvider">The provider used to load runtime configuration.</param>
+        /// <seealso cref="https://docs.microsoft.com/en-us/azure/azure-monitor/app/asp-net-core#enable-application-insights-telemetry-collection"/>
+        private void ConfigureApplicationInsightsTelemetry(IApplicationBuilder app, RuntimeConfigProvider runtimeConfigurationProvider)
+        {
+            if (runtimeConfigurationProvider.TryGetConfig(out RuntimeConfig? runtimeConfig) && runtimeConfig.Runtime.Telemetry is not null
+                && runtimeConfig.Runtime.Telemetry.ApplicationInsights.Enabled)
+            {
+                AppInsightsOptions = runtimeConfig.Runtime.Telemetry.ApplicationInsights;
+
+                if (string.IsNullOrWhiteSpace(AppInsightsOptions.ConnectionString))
+                {
+                    _logger.LogError("Logs won't be sent to Application Insights as connection string is not set in the runtime config.");
+                    return;
+                }
+
+                TelemetryClient? appTelemetryClient = app.ApplicationServices.GetService<TelemetryClient>();
+
+                if (appTelemetryClient is null)
+                {
+                    _logger.LogError("Telemetry client is not initialized.");
```

Comment:
Acc to RFC, Since user enabled application insights, this should be properly initialized to send the telemetry, otherwise error shown.

Reply:
this should be error, if Application Insights is enabled and connection string is also provided but telemetry client in not initialized. this can lead to loss of telemetry, hence error.